File: test_data/java_programs/openjdk_testsuite/02419.java




public @interface AnnoWithClinitFail {
    Foo f = new Foo();

    String foo();
    String bar() default "bar";

    @AnnoWithClinitFail
    static class C {} 
                      

    class Foo {}
}

@AnnoWithClinitFail
class TestAnnoWithClinitFail { }

File: test_data/c_programs/gcc_testsuite/02419.c




struct A
{
  int t;
  int i;
};

struct B
{
  int *p;
  float b;
};

float X;

foo (struct B b, struct A *q, float *h)
{
  X += *h;
  *(b.p) = 3;
  q->t = 2;
  return *(b.p);
}

main()
{
  struct A a;
  struct B b;

  b.p = &a.t;
  if (foo (b, &a, &X) == 3)
    abort ();

  return 0;
}
